internal/consumer/service.go

```diff
@@ -6,37 +6,43 @@ import (
 	"log"
 	"sync"
 
+	"github.com/emmaLP/gs-software-onboarding/internal/database"
 	"github.com/emmaLP/gs-software-onboarding/internal/model"
 	"github.com/emmaLP/gs-software-onboarding/pkg/hackernews"
 	"go.uber.org/zap"
 )
 
-type Service struct {
+type service struct {
 	logger          *zap.Logger
 	numberOfWorkers int
 	hnClient        hackernews.Client
+	dbClient        database.Client
+	context         context.Context
```

Comment:
It's recommended to pass `context` as an argument rather than storing in on a struct. [See more](https://go.dev/blog/context-and-structs)

Reply:
Thanks for sending the info over, it was a good read. will get this updated